app/src/sharedTest/java/org/oppia/android/app/devoptions/ForceNetworkTypeActivityTest.kt

```diff
@@ -0,0 +1,190 @@
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.exploration.lightweightcheckpointing.ExplorationStorageModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.platformparameter.PlatformParameterModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.AccessibilityTestRule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [ForceNetworkTypeActivity]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = ForceNetworkTypeActivityTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class ForceNetworkTypeActivityTest {
+
+  @Inject
+  lateinit var context: Context
+
+  @get:Rule
+  val activityTestRule = ActivityTestRule(
+    ForceNetworkTypeActivity::class.java,
+    /* initialTouchMode= */ true,
+    /* launchActivity= */ false
+  )
+
+  @get:Rule
+  val accessibilityTestRule = AccessibilityTestRule()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testForceNetworkTypeActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(createForceNetworkTypeActivityIntent())
+    val title = activityTestRule.activity.title
+
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(title).isEqualTo(context.getString(R.string.force_network_type_activity_title))
+  }
+
+  @Test
+  fun testForceNetworkTypeActivity_forceNetworkTypeFragmentIsDisplayed() {
+    launch(ForceNetworkTypeActivity::class.java).use {
```

Comment:
This launch command is different from the usually used one in all other test files (eg: MarkTopicsCompletedFragmentTest), any particular reason for this change?

Reply:
Yes. Since the 'ForceNetworkTypeActivity' doesn't take any arguments while creating the intent for the activity, we have used the class approach to launch the activity. This was suggested by @anandwana001 in one of my previous PRs.